addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/handler/EleroGroupHandler.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.elerotransmitterstick.handler;
+
+import java.util.ArrayList;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.openhab.binding.elerotransmitterstick.config.EleroGroupConfig;
+import org.openhab.binding.elerotransmitterstick.stick.ResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EleroGroupHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class EleroGroupHandler extends EleroChannelHandler {
+    private final Logger logger = LoggerFactory.getLogger(EleroGroupHandler.class);
+
+    ArrayList<ResponseStatus> stati;
```

Comment:
plural is status :-)

Reply:
I have no reply button on the previous comment, so I reply here. 

As I have written in my comment, this is a workaround (that will be removed once the firmware bug has been fixed). 
The stick api has a 2 bytes for encoding the channelids, every channel being one bit. Unfortunately, elero does not seem to have a SW QA department, so this does not work as described in the API description (but this has been confirmed as a bug and promised to be fixed by elero). 
How is sending a command to several channels simultaneously supposed to work if you only have one channel ID in the thing?

I am wondering what the benefit is of pressing two different things into one thing type?

Reply:
To link this discussion back to my question, I link it [here](https://github.com/openhab/openhab2-addons/pull/2659#discussion_r151423150).

Maybe I should rephrase what I wanted to express. I think your "eleroThing" thing-type does exactly the same as your "eleroGroup" thing-type with the exception that the first only has one channel-id.

So why should we separate this if the "eleroGroup" thing would be sufficient? This is, if you also allow the user to only specify one channel-id.

Reply:
So how do you then know if the thing is a group or a single channel? You start discovery, which finds 5 channels. Now you add a group manually (of the same thing type). This thing would then accept several ids (how do you define that if it has the same type?), the discovered things only one. How is the user supposed to keep an overview on what are actual channels and what are groups?

Reply:
I am sorry that I have to bother you with some more questions regarding how the elero stuff works:

* As far as I understand your discovery service can only discover plain "elero channels", not groups, right? So the user has to add groups manually and also keep track of the channel-ids he adds to them, right?
* Can ONE physical device have MULTIPLE elero-channels?
* Or is it a 1:1 mapping between physical device and elero-channel?
* What is the ESH status channel (from your items file: `String Rollershutter1State  {channel="elerotransmitterstick:elerochannel:0a0a0a0a:1:status" }`) about? The ESH control channel is bidirectional and sets the state according to your response. I do not (yet) understand the value of your additional status channel.
* As far as I understood your devices all have a state, since you have a bidirectional communication you can tell whether a device is online or offline, is that correct?

Grouping should not be done on a thing level. Things represent the state of a physical device, which may have channels that can influence this state. An item linked to such a channel is the entity which is controlled by the user. If the user wants to control multiple devices at once he has to create an item group with items that each link to a channel of a thing.

Given my description about groups above and that users have to manually create their elero-things-with-multiple-elero-channelIds, so they are aware of which thing points to which elero-channelIds, what can you not express with this model?

Elero-Stick (Bridge) [property serial port] -> Elero-Thing (Thing) [prop: channelIds] -> Control channel [type depends on attached actor, for now RollerShutter]

Reply:
Regarding your questions:
- yes
- yes
- no
- it gives information about the rollershutter state (which does not always fit in the control channel, as this is numeric). there are a lot of possible states that the API supports, have a look at the class ResponseStatus.

You can express everything with this, exactly as you can do with what currently is there. But what is the advantage? You would then have the code that derives the response status in the class that only has one id. You would have to define that the same item type can only have one id if is a thing that represents a channel, multiple otherwise. It is no longer visible to the user with what he is dealing, why he can only have one id for the thing at times, more than one at other times. 
But maybe you can tell me what the advantage of having only one thing type is, I have not yet understood that.